patches/api/0399-Support-for-results-of-ItemStacks-in-CraftItemEvent.patch

```diff
@@ -0,0 +1,50 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Doc <nachito94@msn.com>
+Date: Sun, 9 Oct 2022 11:58:38 -0300
+Subject: [PATCH] Support for results of ItemStacks in CraftItemEvent
+
+
+diff --git a/src/main/java/org/bukkit/event/inventory/CraftItemEvent.java b/src/main/java/org/bukkit/event/inventory/CraftItemEvent.java
+index cab13877fb38c361f18ce9a433e98fea689c9722..fef15af229bced8cfe526800ffe0591eecc6e06c 100644
+--- a/src/main/java/org/bukkit/event/inventory/CraftItemEvent.java
++++ b/src/main/java/org/bukkit/event/inventory/CraftItemEvent.java
+@@ -11,6 +11,7 @@ import org.jetbrains.annotations.NotNull;
+  */
+ public class CraftItemEvent extends InventoryClickEvent {
+     private Recipe recipe;
++    private java.util.List<org.bukkit.inventory.ItemStack> itemResults; // Paper
+ 
+     public CraftItemEvent(@NotNull Recipe recipe, @NotNull InventoryView what, @NotNull SlotType type, int slot, @NotNull ClickType click, @NotNull InventoryAction action) {
+         super(what, type, slot, click, action);
+@@ -22,6 +23,31 @@ public class CraftItemEvent extends InventoryClickEvent {
+         this.recipe = recipe;
+     }
+ 
++    // Paper start - add more data in CraftItemEvent
++    public CraftItemEvent(@NotNull Recipe recipe, @NotNull InventoryView what, @NotNull SlotType type, int slot, @NotNull ClickType click, @NotNull InventoryAction action, @NotNull java.util.List<org.bukkit.inventory.ItemStack> itemResults) {
++        super(what, type, slot, click, action);
++        this.recipe = recipe;
++        this.itemResults = itemResults;
++    }
++
++    public CraftItemEvent(@NotNull Recipe recipe, @NotNull InventoryView what, @NotNull SlotType type, int slot, @NotNull ClickType click, @NotNull InventoryAction action, int key, @NotNull java.util.List<org.bukkit.inventory.ItemStack> itemResults) {
++        super(what, type, slot, click, action, key);
++        this.recipe = recipe;
++        this.itemResults = itemResults;
++    }
++
++    /**
++     * @return A list of copies of the items result from the craft.
++     */
++    @NotNull
++    public java.util.List<org.bukkit.inventory.ItemStack> getItemResults() {
++        if (this.itemResults == null) {
++            return java.util.Collections.singletonList(recipe.getResult().clone());
++        }
++        return this.itemResults;
```

Comment:
This should be an immutable list for consistency

Reply:
I mean the first one wrapped by the singleton is already immutable but the last return isn't

Reply:
ohhh i forget.. originally in server make the list immutable but i detele this.. maybe can add the immutable list in constructor of the event for make immutable list.

Reply:
That's up to you :+1:
Also if you _love_ annotation and think that this isn't too obvious you can add the Unmodifiable annot for the api